packages/plugin-mcp/src/mcp/getMcpHandler.ts

```diff
@@ -43,8 +43,18 @@ export const getMCPHandler = (
   const { payload } = req
   const configSchema = configToJSONSchema(payload.config)
 
+  const payloadToolHandler = (
+    handler: NonNullable<NonNullable<PluginMCPServerConfig['mcp']>['tools']>[number]['handler'],
+  ) => {
+    return async function (...args: unknown[]) {
+      // - Add the Payload request object to the handler
+      // - Add back the _extra argument that comes from the underlying MCP handler
+      return await handler(args[0] as Record<string, unknown>, req, args[1])
```

Comment:
args[0] and args[1] is really not very readable.
We should be using arg objects and destructuring them instead.